app/src/main/java/com/eveningoutpost/dexdrip/insulin/LinearTrapezoidInsulin.java

```diff
@@ -37,18 +37,18 @@ public double calculateIOB(double t) {
         else if ((t1 <= time) && (time < t2))
             return 1.0 + 0.5 * max * t1 - max * time;
         else if ((t2 <= time) && (time < t3))
-            return 0.5 * (time - t3) * (time - t3) * max / (t3 - t2);
+            return 0.5 * (t3 - time) * (t3 - time) * max / (t3 - t2);
```

Comment:
as is is taken to square this change seems useless

Reply:
Agreed, except that you never seen in a textbook that the area of a triangle can be calculated as `( -b) * ( -h) / 2`. And you would think it very strange if you ever saw somebody wrote it that way, even if it works. This is medical software and every "i" should be dotted and every "t" should be crossed. (Not sure if that expression translates well, but I just mean we should be extremely picky about every single little thing.)